src/metabase/lib/js.cljs

```diff
@@ -940,131 +1488,228 @@
 (defn ^:export native-query
   "Create a new native query.
 
-  Native in this sense means a pMBQL query with a first stage that is a native query."
+  *Native* in this sense means a pMBQL query where the first stage is `:mbql.stage/native`.
+
+  > **Code health:** Healthy"
   [database-id metadata inner-query]
   (lib.core/native-query (metadataProvider database-id metadata) inner-query))
 
 (defn ^:export with-native-query
-  "Update the raw native query, the first stage must already be a native type.
-   Replaces templates tags"
+  "Update the raw native query. The first stage of `a-query` must already be a native stage.
+
+  Replaces templates tags.
+
+  > **Code health:** Deprecated. This is never called and should be removed if it's not needed."
```

Comment:
Surprising: what happens when the native query is edited in the ui?

Reply:
I was surprised too. But also I was wrong; I must have typo'd my search for it. It's definitely called.